core/modules/wsched/ChunkDisk.h

```diff
@@ -40,30 +40,32 @@
 #include "memman/MemMan.h"
 #include "proto/worker.pb.h"
 #include "wbase/Task.h"
-
+#include "wsched/ChunkTaskCollection.h"
 
 namespace lsst {
 namespace qserv {
 namespace wsched {
 
 /// Limits Tasks to running when resources are available.
 /// TODO: DM-4943 Maybe merge this class into ScanScheduler.
-class ChunkDisk {
+class ChunkDisk : public ChunkTaskCollection {
```

Comment:
Might we test and add other implementations in the future, if yes we could keep and document this interface?